src/Apps/Legion/Components/GraphVisualizer/Hooks/CustomLassoHandler/CustomLassoHandler.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback } from 'react';
+import { Behaviors } from '@antv/graphin';
+import { IEdge, INode } from '@antv/g6';
+import { getDebugLogger } from '../../../../../../Models/Services/Utils';
+import { ICustomLassoHandlerProps } from './CustomLassoHandler.types';
+import { GraphContextActionType } from '../../../../Contexts/GraphContext/GraphContext.types';
+import { useGraphContext } from '../../../../Contexts/GraphContext/GraphContext';
+import { GetNodeIdsFromSelection } from '../../Utils/GraphUtils';
+
+const { LassoSelect } = Behaviors;
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('CustomLassoHandler', debugLogging);
+
+const CustomLassoHandler: React.FC<ICustomLassoHandlerProps> = (_props) => {
```

Comment:
I am a little confused that these components are under "Hooks" folder but implemented as a react component. Shouldn't it be like useCustomLassoHandler function instead?

Reply:
Explained in the other thread but happy to chat more about it.